### ISSUE TITLE
Allow me to prevent certain fields from being modified

### ISSUE THREAD

**Issue by Mosnar** [verbb/formie#1189]
### What are you trying to do?

In some cases, it's important that certain fields not be modified by content authors - particularly for forms with a higher level of customization, such as sample request forms, registration forms etc. In these cases, content authors should still be able to create new fields, adjust labels etc; however, they should not be able to adjust fields that are critical to website functionality. 

### What's your proposed solution?

I propose adding an option to "lock" fields, which would prevent the field from being modified in any potentially harmful way (deleting, changing field type, changing settings, etc).

This doesn't necessarily need to be a secure, permission-based lock, it could simply be a checkbox in the field settings as a warning that the field shouldn't be touched.

**Comment by engram-design** [comment]
I was about to say that we could employ permissions, but that might get messy (which user groups have access to edit the field). But more importantly, trying to figure out how to "unlock" a field that has been locked, who can (if you can't save the field). I'll give it some thought.

**Comment by Mosnar** [comment]
I was thinking anyone with access to edit the form could unlock it. The idea is simply to deter people from accidentally breaking things. Kind of like how you can lock layers in Photoshop. Maybe an alternative solution could be to add a "note" to fields that shows up in the settings so I could say "do not delete!" - though I prefer the lock idea, personally.

**Comment by engram-design** [comment]
Yeah, I was thinking a generic "note" functionality (similar to the alert notice you get when it's a synced field), just so it's customisable, as a "locked" setting is pretty specific. It's also a bit misleading, because it's not truly knocked (as per your suggestion), its just a note that "hey, don't touch this field please".

**Comment by engram-design** [comment]
Added in [4.0.0-beta.9](https://github.com/verbb/formie/releases/tag/4.0.0-beta.9).
